### DISCUSSION TITLE
Current user with Annotated

### DISCUSSION THREAD

**Discussion by ezeparziale** [fastapi/fastapi#9281] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
CurrentUser = Annotated[User, Depends(get_current_user)]


@router.get("/", status_code=status.HTTP_200_OK)
def get_posts(
    db: Session = Depends(get_db),
    current_user: CurrentUser = None,  # <--- None ?
    commons: CommonsDep = None,
) -> list[PostOut]:
    """
    ### Get post list
    """
    pass
```

### Description

I am testing fastapi ==0.95.0 with annontated feature.
I created
```
CurrentUser = Annotated[User, Depends(get_current_user)]
```
But I am using the default value in current_user = None, this is correct?

```
   ...
    current_user: CurrentUser = None,  # <--- None ?
   ...
```

### Operating System

Windows, macOS

### FastAPI Version

0.95.0

### Python Version

3.10.10

**Comment by nzig** [comment, marked as answer]
Since you specify `Depends(get_current_user)`, `current_user` will always get the value returned by `get_current_user()` when `get_posts()` is called by FastAPI. Therefore, setting the default value with `= None` will have no effect within FastAPI.

The only case it will have an effect is if you call `get_posts()` directly in some other code. In that case, if you don't pass a value for `current_user`, Python will apply the default. Note that this is an uncommon case, since usually only FastAPI calls your handlers and you don't call them yourself as normal functions.

Also as a side note, this is most likely also a type error, unless `User` is some union type that allows `None` as a value.
